### PR TITLE
fix: update HistoryLength

### PR DESCRIPTION
**Pull request by nahapetyan-serob**
Updated HistoryLength to *int in ListTasksRequest.
Updated taskstore and handler to omit the history when HistoryLength is 0.
Updated test suites in a2asrv and taskstore to verify zero-value and negative-value edge cases.

### CODE REVIEW

File: a2asrv/handler.go [hunk 1]
```diff
@@ -217,9 +217,9 @@ func (h *defaultRequestHandler) GetTask(ctx context.Context, req *a2a.GetTaskReq
 	if req.HistoryLength != nil {
 		historyLength := *req.HistoryLength
 
-		if historyLength <= 0 {
+		if historyLength == 0 {
 			task.History = []*a2a.Message{}
```
[the remark this answers is not in the document]
Comment (nahapetyan-serob) [reply a2asrv/handler.go:221]: Updated PR description. Will keep as empty slice.

File: a2asrv/taskstore/inmemory.go [hunk 2]
```diff
@@ -277,13 +275,20 @@ func applyPagination(filteredTasks []*storedTask, pageSize int, req *a2a.ListTas
...
 			return nil, err
 		}
-		if req.HistoryLength > 0 && len(taskCopy.History) > req.HistoryLength {
-			taskCopy.History = taskCopy.History[len(taskCopy.History)-req.HistoryLength:]
+		historyLength := defaultHistoryLength
+		if req.HistoryLength != nil {
+			historyLength = *req.HistoryLength
+		}
+		if historyLength == 0 {
+			taskCopy.History = []*a2a.Message{}
```
[the remark this answers is not in the document]
Comment (nahapetyan-serob) [reply a2asrv/taskstore/inmemory.go:289]: Updated PR description. Will keep as empty slice.

File: a2apb/v0/pbconv/to_proto.go [hunk 3]
```diff
@@ -524,15 +524,19 @@ func ToProtoListTasksRequest(request *a2a.ListTasksRequest) (*a2apb.ListTasksReq
...
 		lastUpdatedAfter = timestamppb.New(*request.StatusTimestampAfter)
 	}
-	return &a2apb.ListTasksRequest{
+	pbReq := &a2apb.ListTasksRequest{
 		ContextId:        request.ContextID,
 		Status:           toProtoTaskState(request.Status),
 		PageSize:         int32(request.PageSize),
 		PageToken:        request.PageToken,
-		HistoryLength:    int32(request.HistoryLength),
+		HistoryLength:    int32(*request.HistoryLength),
```
Comment (yarolegovich) [inline a2apb/v0/pbconv/to_proto.go]: this line should have been removed

File: a2asrv/taskstore/inmemory.go [hunk 4]
```diff
@@ -277,13 +275,20 @@ func applyPagination(filteredTasks []*storedTask, pageSize int, req *a2a.ListTas
 
 func toListTasksResult(tasks []*storedTask, req *a2a.ListTasksRequest) ([]*a2a.Task, error) {
 	var result []*a2a.Task
+	const defaultHistoryLength = 100
```
Comment (yarolegovich) [inline a2asrv/taskstore/inmemory.go]: nit: defaultMaxHistoryLength
